bin/test/ce_install_test.py

```diff
@@ -54,3 +54,68 @@ def test_should_match_end_of_context():
 def test_should_honour_root_matches():
     assert filter_match("/a/b/c", fake("a/b/c", "target"))
     assert not filter_match("/b/c", fake("a/b/c", "target"))
+
+
+def test_should_match_all_filters():
+    assert not filter_aggregate(["a", "b"], fake("a", "target"), filter_match_all=True)
+    assert not filter_aggregate(["a", "b"], fake("b", "target"), filter_match_all=True)
+    assert filter_aggregate(["a", "target"], fake("a", "target"), filter_match_all=True)
+
+def test_should_match_any_filter():
+    assert filter_aggregate(["a", "b"], fake("a", "target"), filter_match_all=False)
+    assert filter_aggregate(["a", "b"], fake("b", "target"), filter_match_all=False)
+    assert not filter_aggregate(["a", "b"], fake("c", "target"), filter_match_all=False)
+
+def test_all_should_match_all_filters():
+    test = [
+        fake("a/x", "target"), fake("a/y", "target"), fake("a/z", "target"), fake("a/z", "other"),
+        fake("b/x", "target"), fake("b/y", "target"), fake("b/z", "target"), fake("b/z", "other"),
+        fake("c/x", "target"), fake("c/y", "target"), fake("c/z", "target"), fake("c/z", "other"),
+    ]
+
+    filter1 = ["a", "b"]
+    test1 = list(filter(lambda installable: filter_aggregate(filter1, installable, filter_match_all=True), test))
```

Comment:
This is actually not equivalent. `filter_aggregate` (like `filter_match`) return a bool for whether the installable passes the filter(s). So in this case `test1` would be a list of bools instead of a list of installables. It would need to be `[i for i in test if filter_aggregate(filter1, i, filter_match_all=True)` which is equivalent to what `filter` does.
Plus, this matches what is written in `ce_install.py`, only here we create a list, instead of leaving it as a filter object.

Reply:
Got it! thakns for explaining! :)